app/controllers/events_controller.rb

```diff
@@ -20,7 +29,11 @@ def create
 
     respond_to do |format|
       if @event.save
-        format.html { redirect_to event_url(@event), notice: "Event was successfully created." }
+        format.html do
+          flash[:notice] = "Event was successfully created."
+          redirect_to event_url(@event)
+          # redirect_to events_path(pagy_calendar_url_at(@calendar, @event.start_date))
```

Comment:
i think that to make this redirect line work, we need to define `@calendar` and all the collection here.

if the url params were fixed

```
# bad
http://localhost:3000/events?year_page=10&month_page=10&day_page=5
# good
http://localhost:3000/events?year_page=2023&month_page=10&day_page=5
```

the calendar pagination would be perfect

Reply:
> i think that to make this redirect line work, we need to define @calendar and all the collection here.

Yes, the `@calendar` should be created in the controller

> if the url params were fixed ... the calendar pagination would be perfect

IIUC you would like to use params to get to any page by time, instead by number. 

To me that looks actually more complicated. If you use `pagy_calendar_url_at(@calendar, time)` you get the url instead of tinkering with 3 params.

Am I missing something?